flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1096,14 +1134,18 @@ void handleAsyncException(String message, Throwable exception) {
 		private final AtomicReference<CheckpointingOperation.AsyncCheckpointState> asyncCheckpointState = new AtomicReference<>(
 			CheckpointingOperation.AsyncCheckpointState.RUNNING);
 
+		private final Future<?> bufferPersistingFuture;
```

Comment:
`Future<?>` -> `Future<Collection<StateObject>>`

Reply:
Hm the type is completely irrelevant. Any reason to change it? I actually haven't really understood, why the future returns the stateobjects to begin with (but Roman verified that all writing has already been done).

Reply:
To add, I consider using `StateObject` in `ChannelStateWriter` to be an implementation detail.

Reply:
My previous consideration was that it might improve readability if we can confirm the specific type already. 
I agree with your point that the type is actually irrelevant and we can regard it as internal details of `ChannelStateWriter`. I have no other concerns for this issue.